src/common/types/timestamp.cpp

```diff
@@ -151,10 +151,18 @@ timestamp_t Timestamp::GetCurrentTimestamp() {
 }
 
 timestamp_t Timestamp::FromEpochSeconds(int64_t sec) {
+	auto max_int64 = std::numeric_limits<int64_t>::max();
+	if (max_int64 / Interval::MICROS_PER_MSEC < sec) {
```

Comment:
This won't work for negative numbers, and not sure if this is entirely correct on the boundary case (this might reject a few too many numbers). I would suggest using `TryMultiplyOperator::Operation` here instead.